proxy/ParentYamlConfig.h

```diff
@@ -0,0 +1,106 @@
+/** @file
+
+  Implementation of Parent Yaml Config parsing
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "ProxyConfig.h"
+#include "yaml-cpp/yaml.h"
+
+using namespace std;
+
+struct parent_dest {
+  parent_dest()
+    : port(0),
+      internal(false),
+      secondary_mode(1),
+      parent_is_proxy(true),
+      unavailable_server_retry_responses("503"),
+      max_simple_retries(1),
+      max_unavailable_server_retries(1),
+      go_direct(true){};
+  string dest_domain;
+  string dest_host;
+  string dest_ip;
+  string url_regex;
+  int port;
+  string scheme;
+  string prefix;
+  string suffix;
+  string method;
+  string time;
+  string src_ip;
+  bool internal;
+  string parent;
+  string secondary_parent;
+  int secondary_mode;
+  bool parent_is_proxy;
+  string parent_retry;
+  string unavailable_server_retry_responses;
+  int max_simple_retries;
+  int max_unavailable_server_retries;
+  string round_robin;
+  bool go_direct;
+  string qstring;
+};
+
+class ParentYamlConfig
+{
+public:
+  ParentYamlConfig();
+  ~ParentYamlConfig();
+  int Init(const std::string &content);
+  bool verifyConfig(const parent_dest &p, uint entry);
+
+  const std::string
+  errorMsg()
+  {
+    return error_msg.str();
+  }
+
+  const uint
+  destSize()
+  {
+    return dest_size;
+  }
+
+  const parent_dest *
```

Comment:
What is the type of `parent_configs[i]`? Could you return a `string_view` or `string const&` here?

Reply:
no-longer using this in the PR, outdated.